### ISSUE TITLE
host python leaks into wheel build venv

### ISSUE THREAD

**Issue by nyobe** [jvolkman/rules_pycross#22]
Apparently there is a `__PYVENV_LAUNCHER__` env var that distutils prefers over argv[0] for choosing the python executable to use when creating a venv... (https://bugs.python.org/issue22490)

Since bazel uses the host python in its bootstrap stub script to launch the actual toolchain, the host python path secretly leaks through this env var, and is used for the isolated build venv instead of the toolchain version.

I think `wheel_builder.py` needs to unset it before invoking venv in order for the correct interpreter to be used.

---

```
> # Building a wheel from examples/poetry
> bazel build @example_lock_repo//deps:setproctitle
ModuleNotFoundError: No module named 'setuptools'

> # Examining the temporary wheelbuild venv, notice its symlinked to host python, not hermetic interpreter
> ls -al /var/folders/ml/kq18l87s1p10lv4dk7p6sb140000gn/T/wheelbuild4p0may9y/env/bin/
python3@ -> /Applications/Xcode.app/Contents/Developer/usr/bin/python3
```

I think I encountered this module lookup failure because my host python version is different than the hermetic version, so it looks in the wrong site-packages directory for the build deps.

**Comment by nyobe** [comment]
Patching `wheel_builder.py` to unset this env var resolves the issue:

**Comment by jvolkman** [comment]
I have experience with `__PYVENV_LAUNCHER__` from elsewhere, but I'm struggling to reproduce this particular issue on my mac. Does this work for you?
```
python3 -c "import os; print(os.environ['__PYVENV_LAUNCHER__'])"
```

Trying to figure out why i'm not seeing it set in my environment.

**Comment by nyobe** [comment]
That doesn't, but this does:
```
/usr/bin/python3 -c "import os; print(os.environ['__PYVENV_LAUNCHER__'])"
/Applications/Xcode.app/Contents/Developer/usr/bin/python3

/usr/bin/python3 --version
Python 3.8.2
```

I'm not sure why the stub script is being executed by this version though, `/usr/bin/env python3` picks up the homebrew install that I'd expect to be used 😞

**Comment by jvolkman** [comment]
Ah, the fix for https://github.com/python/cpython/issues/66680 was backported to Python 3.8 but I'm guessing it landed in something past 3.8.2. #23 has your fix.

**Comment by nyobe** [comment]
ah ha, thank you!
